### PR TITLE
Make a full SQL schema

### PR DESCRIPTION
**Pull request by hawkowl**
v54! v54! v54!

Fixes https://github.com/matrix-org/synapse/issues/5297

### CODE REVIEW

**Comment by ara4n** [comment]
hm..... is the idea that we document the schema in the original deltas then, and so keep them around so we can read the SQL comments? (if so, then cool; just want to understand where to look when I wonder wtf a given column or index is up to on a given table, short of guessing based on the python)

File: synapse/storage/schema/full_schemas/README.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Building full schema dumps
+==========================
+
+Postgres
```
Comment (erikjohnston) [inline synapse/storage/schema/full_schemas/README.txt]: Can you add a comment to say that we need to wait for background updates to have completed?

File: synapse/storage/schema/full_schemas/54/stream_positions.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+INSERT INTO background_updates (update_name, progress_json, depends_on) VALUES
+    ('populate_user_directory_process_rooms', '{}', 'populate_user_directory_createtables');
+
+-- Insert all users, if search_all_users is on
+INSERT INTO background_updates (update_name, progress_json, depends_on) VALUES
+    ('populate_user_directory_process_users', '{}', 'populate_user_directory_process_rooms');
+
+-- Clean up staging tables
+INSERT INTO background_updates (update_name, progress_json, depends_on) VALUES
+    ('populate_user_directory_cleanup', '{}', 'populate_user_directory_process_users');
```
Comment (erikjohnston) [inline synapse/storage/schema/full_schemas/54/stream_positions.sql]: Why do we need these? Surely there's nothing to do here for an empty database?

File: synapse/storage/schema/full_schemas/54/full.sql.sqlite [hunk 3]
```diff
@@ -0,0 +1,260 @@
+CREATE TABLE application_services_state( as_id TEXT PRIMARY KEY, state VARCHAR(5), last_txn INTEGER );
```
Comment (erikjohnston) [inline synapse/storage/schema/full_schemas/54/full.sql.sqlite]: Would be good to just have a comment at the top to say these are auto-generated files and people should look at the delta files for comments

File: synapse/storage/schema/full_schemas/54/stream_positions.sql (hunk 2 above)
Comment (hawkowl) [reply synapse/storage/schema/full_schemas/54/stream_positions.sql]: I wasn't sure if any of them created database tables that other code needed.

File: synapse/storage/schema/full_schemas/README.txt (hunk 1 above)
Comment (hawkowl) [reply synapse/storage/schema/full_schemas/README.txt]: Done

**Comment by hawkowl** [comment]
Tested against the script, and it starts up.

**Pull request merged**